core/rawdb/accessors_chain.go

```diff
@@ -736,16 +735,6 @@ func ReadLogs(db ethdb.Reader, hash common.Hash, number uint64, config *params.C
 		log.Error("Invalid receipt array RLP", "hash", hash, "err", err)
 		return nil
 	}
-
-	body := ReadBody(db, hash, number)
-	if body == nil {
-		log.Error("Missing body but have receipt", "hash", hash, "number", number)
-		return nil
-	}
-	if err := deriveLogFields(receipts, hash, number, body.Transactions); err != nil {
-		log.Error("Failed to derive block receipts fields", "hash", hash, "number", number, "err", err)
-		return nil
-	}
```

Comment:
Review note: need to check for other uses of `ReadLogs`, which could be affected by not deriving the block context fields anymore.